### DISCUSSION TITLE
Pros and Cons of placing mosparo on the different domain

### DISCUSSION THREAD

**Discussion by 0MAL** [mosparo/mosparo#203] [General]
Hi, I am wondering why mosparo should be on the different domain.
I would appreciate if anyone could explain this. Thanks.

**Comment by zepich** [comment]
Hi @0MAL 

Thank you very much for your question. 

We recommend installing mosparo on a different domain (the idea and recommendation is mainly a subdomain. like mosparo.example.com, not a completely different domain) because, with that, you have a clean virtual host and can put the document root of that virtual host to the `/public` directory of mosparo.

The idea is that all the 'internal' directories, like the `/var/cache` directory or the `/config` directory, are not exposed to the public. We do our best to make it as secure as possible, but it's hard to catch all the possible cases (for example: the `.htaccess` file works great to limit access, but does not work at all with nginx as webserver).

Because of this risk, we recommend using a custom domain (subdomain or completely new domain) to have a virtual host just for this domain and mosparo, where you can set the document root directly to the `/public` directory of mosparo. With this approach, the risk of someone accessing a sensitive or problematic file is very low.

Of course, you can install it in a subdirectory of your main document root. The `.htaccess` file rewrites all the paths to the public directory, and everything should work as expected.

Please let me know if you have more questions.

Kind regards,

zepich

**Comment by 0MAL** [comment]
Thank you. I understand that.
BTW, when the name of public folder for subdomains is public_html, in which files should I modify the path?

**Reply by zepich** [reply]
Hi @0MAL 

Thank you very much for your question.

I would recommend putting all the mosparo code into the `public_html` directory and then changing your host's document root from `/public_html` to `/public_html/public`.

If you change the mosparo code directly and install an update, the update will revert all your changes.

But if you want to adjust the path from `/public` to `/public_html`, you have to edit at least the file src/Helper/DesignHelper and change the public paths there to `public_html` (https://github.com/mosparo/mosparo/blob/c9187c5c84492c373db562f16f6d1db45f944093/src/Helper/DesignHelper.php#L271 and https://github.com/mosparo/mosparo/blob/c9187c5c84492c373db562f16f6d1db45f944093/src/Helper/DesignHelper.php#L299)

I haven't tested this, so I cannot guarantee that everything will work after that.

Please let me know if you need more help.

Kind regards,

zepich

**Comment by 0MAL** [comment]
Ok, thank you.
